### PR TITLE
WM4_Dawit-Abraha_JavaScript_Core-1_Week-1

### PR DESCRIPTION
**Pull request by Dawit-Dev**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Dawit Abraha
- Your City: Coventry
- Your Slack Name: Dawit Abraha

# Homework Details

- Module: JavaScript_Core-1
- Week: 1

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

-----
[View rendered exercises/C-variables/README.md](https://github.com/Dawit-Dev/JavaScript-Core-1-Coursework-Week1/blob/master/exercises/C-variables/README.md)

### CODE REVIEW

**Review by Mesgna616 — COMMENTED** [review]
Dawit, your codes are neat and easy to read and understand them. And the way you tackle the problems is good and i like analytical approach.

File: exercises/J-functions/exercise2.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
 function triple(number) {
   // complete function here
+  var number = 12;
```
Comment (kawaMANMI) [inline exercises/J-functions/exercise2.js:3]: I think you don't need write step 3  (var number = 12;)  as passed "number" as parameter in the function.

**Review by kawaMANMI — COMMENTED** [review]
Well done Dawit! overall very good starting in JavaScript

File: exercises/F-strings-methods/exercise.js [hunk 2]
```diff
@@ -1,3 +1,8 @@
 // Start by creating a variable `message`
-
+var name = "Dawit";
+var messageStart = "My name is";
+var messageMiddle = "and my name is";
+var messageEnd = "charactors long";
+var nameLength = name.nameLength;
```
[the remark this answers is not in the document]
Comment (Dawit-Dev) [reply exercises/F-strings-methods/exercise.js:6]: I appreciate the comments. I'll make it right.

File: mandatory/3-function-output.js [hunk 3]
```diff
@@ -2,16 +2,24 @@
...
-
+ //*
+// The random() method returns a random value that is greater than or equal to 0.0 and less than 1.0.
+//*
 // Add comments to explain what this function does. You're meant to use Google!
 function combine2Words(word1, word2) {
   return word1.concat(word2);
 }
+//*
+// This function will join the two words without space between them.
```
Comment (jdbevan) [inline mandatory/3-function-output.js:13]:
```suggestion
// This function will join the two words with a space between them.
```

File: mandatory/3-function-output.js [hunk 4]
```diff
@@ -2,16 +2,24 @@
 function getRandomNumber() {
   return Math.random() * 10;
 }
-
+ //*
+// The random() method returns a random value that is greater than or equal to 0.0 and less than 1.0.
```
Comment (jdbevan) [inline mandatory/3-function-output.js:6]: Nice explanation!

File: mandatory/3-function-output.js [hunk 5]
```diff
@@ -2,16 +2,24 @@
...
   return word1.concat(word2);
 }
+//*
+// This function will join the two words without space between them.
+//*
 
 function concatenate(firstWord, secondWord, thirdWord) {
   // Write the body of this function to concatenate three words together.
   // Look at the test case below to understand what this function is expected to return.
+  return firstWord.concat(" ", secondWord).concat(" ", thirdWord);
```
Comment (jdbevan) [inline mandatory/3-function-output.js:19]: I like the way you've called `.concat()` on the result of `.concat()`! Are you able to explain why this works?

File: mandatory/4-tax.js [hunk 6]
```diff
@@ -5,7 +5,11 @@
   Sales tax is 20% of the price of the product.
 */
 
-function calculateSalesTax() {}
+function calculateSalesTax(price) {
+  let salesTax = (price * 20) / 100;
```
Comment (jdbevan) [inline mandatory/4-tax.js:9]: This confused me to begin with as to why you were multiplying the price by 20. One of the reasons is that I have no idea what the significance of that number is. In the industry we'd refer to this as a "magic number". It's considered better to treat it as a named constant defined somewhere in your code. Some people/organisations will capitalize shared constants, something like:
```
const TAX_RATE = 20;

function calculateSalesTax(price) {
    let tax = price * TAX_RATE / 100;
    return price + tax;
}
```

**Pull request closed without merging**